src/tsickle.ts

```diff
@@ -853,13 +856,20 @@ class Annotator extends ClosureRewriter {
       }
       const resolved = ts.resolveModuleName(moduleId, this.file.fileName, this.tsOpts, this.host);
       if (resolved && resolved.resolvedModule) {
-        const resolvedModule = resolved.resolvedModule.resolvedFileName.replace(/(\.d)?\.ts$/, '');
-        const requestedModule = moduleId.replace(/\.js$/, '');
+        const requestedModule = moduleId.replace(EXT, '');
+        const resolvedModule = resolved.resolvedModule.resolvedFileName.replace(EXT, '');
         // If the imported module resolves to foo/index, but the specified module was foo, then we
         // append the /index.
-        if (resolvedModule.substr(resolvedModule.length - 6) === '/index' &&
-            requestedModule.substr(requestedModule.length - 6) !== '/index') {
-          moduleId += '/index';
+        if (requestedModule.substr(requestedModule.lastIndexOf('/')) !==
+            resolvedModule.substr(resolvedModule.lastIndexOf('/'))) {
+          moduleId = path.relative(this.file.fileName, resolvedModule)
+                         // path.relative starts from the parent directory (../self) but ES6
+                         // imports are relative to the current directory (./self) so we strip
+                         // one dot.
```

Comment:
Oh I see, it treats the first argument as if it's a directory.  Maybe it'd be more readable as
```js
path.relative(path.dirname(this.file.fileName), resolveModule)
```
and then drop the regex??